### ISSUE TITLE
[Error when testing on custom data]

### ISSUE THREAD

**Issue by csikasote** [facebookresearch/flashlight#391]
### Question
I am getting an error when testing on custom data when following the tutorial (https://github.com/facebookresearch/flashlight/tree/master/flashlight/app/asr/tutorial) to finetune on custom data. 

Command
--------------------------------
 ./flashlight/build/bin/asr/fl_asr_test --am model.bin --datadir '' --emission_dir '' --uselexicon false \
            --test /content/drive/MyDrive/wav2letter/datastore_dir/wav2letter_dts/bembaspeech/lists/test.lst \
            --tokens /content/drive/MyDrive/wav2letter/datastore_dir/wav2letter_dts/bembaspeech/text/tokens.txt \
            --lexicon /content/drive/MyDrive/wav2letter/datastore_dir/wav2letter_dts/bembaspeech/text/lexicon.txt \
            --show 

The Error is as following:
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------
I0104 11:57:26.232728  9665 Test.cpp:109] Gflags after parsing 
--flagfile=; --fromenv=; --tryfromenv=; --undefok=; --tab_completion_columns=80; --tab_completion_word=; --help=false; --helpfull=false; --helpmatch=; --helpon=; --helppackage=false; --helpshort=false; --helpxml=false; --version=false; --adambeta1=0.94999999999999996; --adambeta2=0.98999999999999999; --am=model.bin; --am_decoder_tr_dropout=0.20000000000000001; --am_decoder_tr_layerdrop=0.20000000000000001; --am_decoder_tr_layers=6; --arch=EG_GLU1x2048_S3_TR36x384_1536_DO0.05_LD0.05_CTC; --attention=keyvalue; --attentionthreshold=2147483647; --attnWindow=softPretrain; --attnconvchannel=0; --attnconvkernel=0; --attndim=0; --batching_max_duration=0; --batching_strategy=none; --batchsize=4; --beamsize=2500; --beamsizetoken=250000; --beamthreshold=25; --channels=1; --criterion=ctc; --critoptim=adagrad; --datadir=; --decoderattnround=1; --decoderdropout=0; --decoderrnnlayer=1; --decodertype=wrd; --devwin=0; --emission_dir=; --emission_queue_size=3000; --enable_distributed=true; --encoderdim=256; --eosscore=0; --eostoken=false; --everstoredb=false; --fftcachesize=1; --filterbanks=80; --fl_amp_max_scale_factor=32000; --fl_amp_scale_factor=4096; --fl_amp_scale_factor_update_interval=2000; --fl_amp_use_mixed_precision=false; --fl_benchmark_mode=true; --fl_log_level=; --fl_optim_mode=; --fl_vlog_level=0; --flagsfile=; --framesizems=25; --framestridems=10; --gamma=1; --gumbeltemperature=1; --highfreqfilterbank=-1; --inputfeeding=false; --isbeamdump=false; --iter=9223372036854775807; --itersave=false; --labelsmooth=0.050000000000000003; --leftWindowSize=50; --lexicon=/content/drive/MyDrive/wav2letter/datastore_dir/wav2letter_dts/bembaspeech/text/lexicon.txt; --linlr=-1; --linlrcrit=-1; --linseg=0; --lm=; --lm_memory=5000; --lm_vocab=; --lmtype=kenlm; --lmweight=0; --lmweight_high=4; --lmweight_low=0; --lmweight_step=0.20000000000000001; --localnrmlleftctx=0; --localnrmlrightctx=0; --logadd=false; --lowfreqfilterbank=0; --lr=0.02; --lr_decay=100; --lr_decay_step=40; --lrcosine=false; --lrcrit=0.02; --max_devices_per_node=8; --maxdecoderoutputlen=400; --maxgradnorm=0.10000000000000001; --maxload=-1; --maxrate=10; --maxsil=50; --maxword=-1; --melfloor=1; --mfcc=false; --mfcccoeffs=13; --mfsc=true; --minrate=3; --minsil=0; --momentum=0.94999999999999996; --netoptim=adagrad; --nthread=6; --nthread_decoder=1; --nthread_decoder_am_forward=1; --numattnhead=8; --onorm=target; --optimepsilon=1e-08; --optimrho=0.90000000000000002; --pctteacherforcing=99; --pcttraineval=1; --pow=false; --pretrainWindow=0; --replabel=0; --reportiters=4000; --rightWindowSize=50; --rndv_filepath=; --rundir=/checkpoint/padentomasello/wav2letter_experiments/256_GPU; --samplerate=16000; --sampletarget=0.01; --samplingstrategy=rand; --saug_fmaskf=30; --saug_fmaskn=2; --saug_start_update=24000; --saug_tmaskn=10; --saug_tmaskp=0.050000000000000003; --saug_tmaskt=30; --sclite=; --seed=0; --sfx_config=; --show=true; --showletters=false; --silscore=0; --smearing=none; --smoothingtemperature=1; --softwoffset=10; --softwrate=5; --softwstd=4; --sqnorm=true; --stepsize=9223372036854775807; --surround=; --test=/content/drive/MyDrive/wav2letter/datastore_dir/wav2letter_dts/bembaspeech/lists/test.lst; --tokens=/content/drive/MyDrive/wav2letter/datastore_dir/wav2letter_dts/bembaspeech/text/tokens.txt; --train=/private/home/qiantong/wav2letter_experiments/256_GPU/join_train_notl_maxlen.25s.lst; --trainWithWindow=true; --transdiag=0; --unkscore=-inf; --use_memcache=false; --uselexicon=true; --usewordpiece=false; --valid=dev-clean:/checkpoint/jacobkahn/data/lists/librispeech/dev-clean.lst,dev-other:/checkpoint/jacobkahn/data/lists/librispeech/dev-other.lst,test-clean:/checkpoint/jacobkahn/data/lists/librispeech/test-clean.lst,test-other:/checkpoint/jacobkahn/data/lists/librispeech/test-other.lst,hub05_callhome:/checkpoint/wav2letter/data/swbd_lists/noNL/hub05-callhome.16khz.lst,hub05_switchboard:/checkpoint/wav2letter/data/swbd_lists/noNL/hub05-switchboard.16khz.lst,commonvoice_dev:/checkpoint/wav2letter/data/commonvoice/lists/dev.lst,commonvoice_test:/checkpoint/wav2letter/data/commonvoice/lists/test.lst,nov92:/checkpoint/qiantong/wsj/nov92.lst; --validbatchsize=-1; --warmup=48000; --weightdecay=0; --wordscore=0; --wordseparator=|; --world_rank=0; --world_size=1; --alsologtoemail=; --alsologtostderr=false; --colorlogtostderr=false; --drop_log_memory=true; --log_backtrace_at=; --log_dir=; --log_link=; --log_prefix=true; --logbuflevel=0; --logbufsecs=30; --logemaillevel=999; --logfile_mode=436; --logmailer=/bin/mail; --logtostderr=true; --max_log_size=1800; --minloglevel=0; --stderrthreshold=2; --stop_logging_if_full_disk=false; --symbolize_stacktrace=true; --v=0; --vmodule=; 
I0104 11:57:26.235389  9665 Test.cpp:130] Number of classes (network): 29
I0104 11:57:26.274847  9665 Test.cpp:137] Number of words: 27210
terminate called after throwing an instance of 'std::invalid_argument'
  what():  stof
*** Aborted at 1609761446 (unix time) try "date -d @1609761446" if you are using GNU date ***
PC: @     0x7feb8f962fb7 gsignal
*** SIGABRT (@0x25c1) received by PID 9665 (TID 0x7febdc099780) from PID 9665; stack trace: ***
    @     0x7febd4e76980 (unknown)
    @     0x7feb8f962fb7 gsignal
    @     0x7feb8f964921 abort
    @     0x7febb2a686f5 __gnu_cxx::__verbose_terminate_handler()
    @     0x7febb29d9b36 __cxxabiv1::__terminate()
    @     0x7febb29d9b81 std::terminate()
    @     0x7febb29cd933 __cxa_throw
    @     0x7febb2a5d35f std::__throw_invalid_argument()
    @     0x55db3359270c fl::app::asr::ListFileDataset::ListFileDataset()
    @     0x55db3356d06f _ZN2fl3app3asr13createDatasetERKSt6vectorINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEESaIS8_EERKS8_iRKSt8functionIFN2af5arrayEPvNSG_4dim4E8af_dtypeEESO_SO_RKSt5tupleIJiiiEEiibSE_i
    @     0x55db333c0969 main
    @     0x7feb8f945bf7 __libc_start_main
    @     0x55db3342ca8a _start

#### Additional Context
The AMI dataset in the colab is running FINE without any problem.  
I generated all the required files (*.list, *.txt) following the LibriSpeech recipe considering that my dataset is formated similar to LibriSpeech dataset. See my files here:

Lists: https://drive.google.com/drive/folders/1fq-j0dGFKEfAfSIpmm7nN2vrFkllG1Aw?usp=sharing
Texts: https://drive.google.com/drive/folders/1fh8KAvbQ7cSA-WI3AdIklpTM_aPIGr8i?usp=sharing

**Comment by csikasote** [comment]
I figured it out. I had spaces in the path lines. Case closed!
